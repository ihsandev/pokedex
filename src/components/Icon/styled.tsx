import styled from "styled-components";

export const IconStyle = styled.div`
  font-size: ${({ size }: any) => size}px;
  color: ${({ color }: any) => color};
`;
