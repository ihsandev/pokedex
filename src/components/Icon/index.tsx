import React from "react";
import { IconStyle } from "./styled";

export interface Props {
  name?: string;
  size?: number;
  color?: string;
}
const Icon: React.FC<Props> = ({ name, ...props }) => {
  return (
    <IconStyle {...props}>
      <i className={name} />
    </IconStyle>
  );
};

export default Icon;
