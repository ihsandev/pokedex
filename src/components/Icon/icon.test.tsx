import * as React from "react";
import { shallow } from "enzyme";
import Icon from "./index";

test("render Icon with props", () => {
  const wrapper = shallow(
    <Icon name="ri-delete-bin-2-line" color="red" size={30} />
  );
  expect(wrapper).toMatchSnapshot();
});
