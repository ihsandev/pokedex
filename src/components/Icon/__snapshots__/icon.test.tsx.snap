// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Icon with props 1`] = `
<styled.div
  color="red"
  size={30}
>
  <i
    className="ri-delete-bin-2-line"
  />
</styled.div>
`;
